src/lib.rs

```diff
@@ -584,4 +584,45 @@ mod tests {
         client.read_to_end(&mut bytes).unwrap();
         assert_eq!(&bytes, b"bye world");
     }
+
+    // futures channels can exchange information between different threads
+    #[test]
+    fn interthread_communication() {
+        use futures::sync::oneshot;
+
+        let (tx0, rx0) = oneshot::channel::<u32>();
+        let (tx1, rx1) = oneshot::channel::<u32>();
+
+        let rt = Runtime::new();
```

Comment:
I think this is specifically testing the case where there are multiple threads, so maybe set `pool_size(2)` to ensure that?

Reply:
Good point.